primitives/src/json.rs

```diff
@@ -0,0 +1,166 @@
+//! String containing serialized valid JSON.
+//! This string is guaranteed to parse as JSON
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    borrow::ToOwned,
+    string::{String, ToString},
+};
+use core::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+#[cfg(feature = "std")]
+use std::string::{String, ToString};
+
+use iroha_schema::IntoSchema;
+use parity_scale_codec::{Decode, Encode};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json::Value;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Eq, Encode, Decode)]
+pub struct JsonString(pub String);
+
+/// A candidate for a valid `JsonString`.
+/// Is used for generalizing ser/de any types to `JsonString` and vise versa
+#[derive(Serialize, Deserialize, Clone)]
+pub struct JsonCandidate<T>(pub T);
```

Comment:
make it private. We usually put it into `mod candidate`. Take a look at `block.rs` for example

Reply:
I can see you're using it in one test, but you really shouldn't

Reply:
> make it private. We usually put it into `mod candidate`. Take a look at `block.rs` for example

I've made it private, but there is no need to put it into separate module